extension/observer/endpoints.go

```diff
@@ -54,6 +123,23 @@ type Port struct {
 	Transport Transport
 }
 
+func (p *Port) Env() EndpointEnv {
+	return map[string]interface{}{
+		"name": p.Name,
+		"port": p.Port,
+		"pod": map[string]interface{}{
+			"name":        p.Pod.Name,
+			"labels":      p.Pod.Labels,
+			"annotations": p.Pod.Annotations,
+		},
+		"transport": p.Transport,
+	}
+}
+
+func (p *Port) Type() EndpointType {
+	return PortType
+}
+
 // HostPort is an endpoint discovered on a host.
```

Comment:
Having trouble following what a "host port" is.  is Port only applicable for containers?

Reply:
Yeah, the type needs to be unique since you can have multiple observers being watched and so you don't want to call it "port" because that would match container ports as well. Which begs the question why is container port just called port which is it was first before I realized the problem. :-P

Reply:
Could be worth renaming, since this becomes potentially more confusing when the docker observer lands?  ContainerPort and ProcessPort?

Reply:
I've filed #2015. There's other possible breaking changes to be made around type so would want to do them all at once.

Reply:
@jrcamp so you saying "port" will be renamed to "containerport" and "hostport" will become just "port"?

Reply:
I think it'd stay hostport to make it unambiguous. But containerport should really be something like k8scontainerport because you could have a docker observer with its own containerport type with different attributes. Starts to get kinda gross because your rules start looking like `type.k8scontainerport && ...`.

This is all because a receiver_creator instance can watch different types of observers in the same instance (ie, it can watch a host observer and a k8s observer simultaneously). Maybe instead we limit it to watching the same types of observers (if you want to watch multiple types you just add another receiver_creator instance). This gets rid of the mixed type confusion and we can just call a hostport and containerport port.